engine/physics/src/physics/physics_3d.cpp

```diff
@@ -424,9 +424,12 @@ namespace dmPhysics
 
         {
             DM_PROFILE(Physics, "StepSimulation");
-            // Step simulation
-            // TODO: Max substeps = 1 for now...
-            world->m_DynamicsWorld->stepSimulation(dt, fixed_time_step ? 0 : 1);
+
+            // We want to make sure that if necessary, we at least cover two (potential) updates
+            // in order to not truncate the time (it would make the simulation go slower)
+            int maxSteps = 2;
+
+            world->m_DynamicsWorld->stepSimulation(dt, maxSteps, fixed_timestep);
```

Comment:
The main problem I have is that I just couldn't get 30Hz mode to work the same way as the 60Hz mode.
The arguments suggests that it should work, but the resulting velocity due to gravity always ends up faster than desired. 🤔 

I'll just have to leave that for later, until we come up with some new ideas.

(60hz seems like a good standard anyways, the default value in Godot as well)

Reply:
Unfortunately, no. Those are the same settings as in 2.77 and they are describe in the same manner in this version.
I think I've played with every possible combination of these settings, to no avail.
I considered exposing these settings as "buller.max_steps" etc, but since I couldn't get them to work the way I wanted it to it felt unnecessary at this point.

At one point, it would be good to update to 2.83 though (preferably when moving it out into an extension).

Reply:
I've tested this branch and the 3D physics behaviour with different arguments this morning. Now, I see the point. Yes.

I think that you should apply this PR and keep it as is. Plus, add a note to the Defold manual - "Use fixed update frequency = 60 and fixed timestep for 3D physics to have the best results". Something like that.

Reply:
Yes, it's strange.

In this last commit, I did add the "max_fixed_timesteps" that can also be used to control the behavior. (default is 2)
This way, the user, given knowledge about how bullet works, can control the call to the `stepSImulation()`

Your suggestion for the documentation is good as well, and I'll make sure to update it